weave/trait/telemetry/wdm/telemetry_wdm_trait.proto

```diff
@@ -0,0 +1,94 @@
+//
+//    Copyright (c) 2016 Nest Labs, Inc.
+//    All rights reserved.
+//
+//    This document is the property of Nest. It is considered
+//    confidential and proprietary information.
+//
+//    This document may not be reproduced or transmitted in any form,
+//    in whole or in part, without the express written permission of
+//    Nest.
+//
+
+//    @file
+//      This file specifies a WDM Telemetry trait that specifies the
+//      various telemetry counters that are associated with WDM
+
+syntax = "proto3";
+
+package weave.trait.telemetry.wdm;
+
+import "wdl/wdl_options.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "WeaveInternalTelemetryWdmTrait";
+option objc_class_prefix = "SCM";
+
+//  @brief
+//    TelemetryWdmTrait
+//
+//  A telemetry trait for the WDM that specifies various counters
+//  associated with WDM updates and subscriptions.
+//
+message TelemetryWdmTrait {
+  option (wdl.message_type) = TRAIT;
+  option (wdl.trait) = {
+    stability: ALPHA,
+    id: 0x1781,
+    vendor_id: 0x0000,
+    version: 1
+  };
+  option (wdl.properties) = {
+    writable: READ_ONLY,
+  };
+
+  // Events associated with WDM updates
+  message TelemetryWdmUpdateStatsEvent {
```

Comment:
Would it make sense to log the update and subscription failures as first class events rather than remembering the last event in a different event?  The semantics of most of the field in this trait make quite a lot of sense as properties.  As events I would suggest we expose 3 different events: subscription failure (fields: reason), update failure (fields: reason, profile, status_code), and counts (fields: successful_updates, unsuccessful_updates, successful_subscriptions, subscription_attempts).  If desired, the last subscription failure and last update failure could also be exposed as properties, so they are available for inspection on connectivity recovery

Reply:
Yes, I think the last failures would better serve as trait properties. That makes sense especially if these events are not uploaded periodically and visibility of the last failures might fade unless preserved as properties.
If we break these into multiple events, then the subscription and update failures could be triggered on the edges while the counter is a periodic one with a relatively long cadence(~20-30 mins)

Reply:
Modified to what was discussed offline , update on edge events for failure with properties in case the events are dropped lost due to lack of buffer space.